src/services/sync-component-with-file/find-config-file-changes.test.ts

```diff
@@ -112,4 +105,135 @@ describe('findConfigAsCodeFileChanges', () => {
 
     expect(result).toEqual(expectedResult);
   });
+
+  describe('unlink component from modified file by immutableLocalKey', () => {
+    let event: PushEvent;
+    const MODIFIED_BEFORE = 'modifiedBefore';
+    const MODIFIED_AFTER = 'modifiedAfter1';
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      getCommitDiffMock.mockResolvedValue([commitDiffMock]);
+      event = generatePushEvent();
+      process.env.CREATE_FROM_YAML_FF = 'false';
+    });
+
+    it('unlinks component when ID changed to immutableLocalKey', async () => {
+      process.env.CREATE_FROM_YAML_FF = 'true';
+      const modifiedMock = [createCommitFileDiffMock([{ id: MODIFIED_BEFORE }, { immutableLocalKey: MODIFIED_AFTER }])];
+
+      groupDiffsByChangeTypeMock.mockReturnValue({
+        added: [],
+        modified: modifiedMock,
+        removed: [],
+      });
+      const expectedResult: ComponentChanges = {
+        componentsToSync: [{ componentYaml: { immutableLocalKey: MODIFIED_AFTER }, absoluteFilePath: 'new/path' }],
+        componentsToUnlink: [{ id: MODIFIED_BEFORE }],
+      };
+
+      const result = await findConfigAsCodeFileChanges(event, 'token');
+
+      expect(result).toEqual(expectedResult);
+    });
+
+    it('unlinks component when ID changed to immutableLocalKey and FF turned off', async () => {
+      process.env.CREATE_FROM_YAML_FF = 'true';
```

Comment:
based on the test title, this should be off

Reply:
Oh, good catch!
I've changed the value to "false". The test is left the same because the component should be unlinked when the ID changes in both cases.